pkg/trace/agent/agent.go

```diff
@@ -196,9 +196,10 @@ func (a *Agent) Process(p *api.Payload) {
 	defer timing.Since("datadog.trace_agent.internal.process_payload_ms", time.Now())
 	ts := p.Source
 	ss := new(writer.SampledChunks)
-	var envtraces []stats.EnvTrace
 	a.PrioritySampler.CountClientDroppedP0s(p.ClientDroppedP0s)
+	statsInput := stats.NewStatsInput(p.TracerPayload, p.ClientComputedStats, a.conf)
```

Comment:
nit: Reading this file, it seems concerning that you are passing in the `TracerPayload` to `NewStatsInput` and then immediately mutating it. It would improve readability if we just pass in the fields from TracerPayload that we need (ContainerID, numChunks).